### PR TITLE
enable python tests in cmake

### PR DESCRIPTION
**Pull request by noskill**
enable python tests in cmake

fix python errors decoding for python3

Should partially fix python tests https://github.com/opencog/atomspace/issues/1801 , i have only one test failing PythonEvalUTest

### CODE REVIEW

**Comment by noskill** [comment]
I made new issue for remaining test https://github.com/opencog/atomspace/issues/1888

File: CMakeLists.txt [hunk 1]
```diff
@@ -544,10 +544,7 @@ IF (CXXTEST_FOUND)
 		)
 	ENDIF (CMAKE_BUILD_TYPE STREQUAL "Coverage")
 	ADD_CUSTOM_TARGET(test_python
-		DEPENDS PythonEvalUTest
-				PythonUtilitiesUTest
-				bindlink_cython atomspace_cython
-				scheme_wrapper logger_cython utilities_cython
+		DEPENDS tests
```
Comment (vsbogd) [inline CMakeLists.txt]: Why test_python depends on tests? 
Should it be just removed instead?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -39,9 +39,10 @@ IF (CXXTEST_FOUND)
 	# Reasoning depends on query, above.
 	ADD_SUBDIRECTORY (rule-engine)
 
-	IF (HAVE_CYTHON AND PYTHONINTERP_FOUND)
+	IF (HAVE_CYTHON AND HAVE_NOSETESTS)
+		MESSAGE(STATUS "CYTHON")
```
Comment (vsbogd) [inline tests/CMakeLists.txt]: I would add more verbose message like "Cython found, enabling Cython unit tests"

**Comment by vsbogd** [comment]
Please ignore the tests which are failed (PythonEvalUTest as I understand) https://github.com/opencog/atomspace/issues/1888

File: CMakeLists.txt (hunk 1 above)
Comment (noskill) [reply CMakeLists.txt]: @inflector Do you think we can remove test_python and run python tests together with all other tests?

File: CMakeLists.txt [hunk 3]
```diff
@@ -544,10 +544,7 @@ IF (CXXTEST_FOUND)
 		)
 	ENDIF (CMAKE_BUILD_TYPE STREQUAL "Coverage")
 	ADD_CUSTOM_TARGET(test_python
-		DEPENDS PythonEvalUTest
-				PythonUtilitiesUTest
-				bindlink_cython atomspace_cython
-				scheme_wrapper logger_cython utilities_cython
+		DEPENDS tests PythonEval logger_cython scheme_wrapper bindlink_cython
```
Comment (linas) [inline CMakeLists.txt]: This seems to be a misunderstanding. The original intent was to be able to say `make test_python` and as a result, ONLY the python-related tests would run. This can save a lot of time during development, by avoiding the overhead of re-running all of the other tests.   This change seems to completely wreck the intended behavior.

Comment (linas) [reply CMakeLists.txt]: Just to be clear: if one says `make test`, then all of the python tests should *also* run -- The `make test_python` is just a short-cut, to run a subset of all of the unit tests.

Comment (noskill) [reply CMakeLists.txt]: Thanks Linas, there were some misunderstanding on my side, now make test should successfully run python tests, i updated dependencies

File: tests/cython/CMakeLists.txt [hunk 4]
```diff
@@ -26,20 +26,20 @@ LINK_LIBRARIES(
...
+#	clearbox
+#	PythonEval
+#	atomspace_cython
+#	${COGUTIL_LIBRARY}
+#)
+
+#SET_TESTS_PROPERTIES(PythonEvalUTest
+#	PROPERTIES
+#	ENVIRONMENT "PYTHONDONTWRITEBYTECODE=1"
+#	ENVIRONMENT "PYTHONPATH=${PROJECT_BINARY_DIR}/opencog/cython")
```
[the remark this answers is not in the document]
Comment (noskill) [reply tests/cython/CMakeLists.txt]: Reverted

**Comment by vsbogd** [comment]
Looks good to me

File: CMakeLists.txt [hunk 5]
```diff
@@ -544,14 +544,12 @@ IF (CXXTEST_FOUND)
 		)
 	ENDIF (CMAKE_BUILD_TYPE STREQUAL "Coverage")
 	ADD_CUSTOM_TARGET(test_python
-		DEPENDS PythonEvalUTest
-				PythonUtilitiesUTest
-				bindlink_cython atomspace_cython
-				scheme_wrapper logger_cython utilities_cython
+		DEPENDS PythonEval logger_cython scheme_wrapper bindlink_cython atomspace_cython
```
Comment (linas) [inline CMakeLists.txt]: PythonEvalUTest and PythonUtilitiesUTest are still missing.  However, PythonEval is not a unit test, so it cannot actually run. 

Did you actually try to run `make test_python` ?  when you say that, it should run all (seven) python tests.

File: CMakeLists.txt [hunk 6]
```diff
@@ -544,14 +544,12 @@ IF (CXXTEST_FOUND)
...
 		WORKING_DIRECTORY tests/cython
 		COMMAND ${CMAKE_CTEST_COMMAND} $(ARGS)
 		COMMENT "Running Python and Cython tests..."
 	)
+	add_dependencies(tests PythonEval logger_cython scheme_wrapper bindlink_cython atomspace_cython)
```
Comment (linas) [inline CMakeLists.txt]: Why is this needed? The tests themselves should already declare the correct dependencies; you should not need to also declare additional dependencies here.

File: CMakeLists.txt (hunk 5 above)
Comment (linas) [reply CMakeLists.txt]: Oh, I see ... OK, dependencies. Yes, I guess these are sqonky.  Never mind; I'll just merge this.

File: CMakeLists.txt (hunk 6 above)
Comment (noskill) [reply CMakeLists.txt]: I didn't figure out how, `add_dependencies` won't work on tests, so I used this workaround.
